libs/utils/WordPressUtils/src/test/java/org/wordpress/android/util/LogFileHelpersTest.kt

```diff
@@ -0,0 +1,64 @@
+package org.wordpress.android.util
+
+import android.content.Context
+import android.os.Build
+import androidx.test.core.app.ApplicationProvider
+import java.io.File
+import java.util.UUID
+import org.junit.After
+import org.junit.Assert
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+import org.robolectric.annotation.Config
+import org.wordpress.android.util.helpers.logfile.LogFileProvider
+
+@RunWith(RobolectricTestRunner::class)
+@Config(sdk = [Build.VERSION_CODES.O_MR1])
+class LogFileHelpersTest {
+    lateinit var testProvider: LogFileProvider
+
+    @Before
+    fun setup() {
+        val context: Context = ApplicationProvider.getApplicationContext()
+        testProvider = LogFileProvider.fromContext(context)
+    }
+
+    @After
+    fun tearDown() {
+        // Delete the test directory after each test
+        testProvider.getLogFileDirectory().deleteRecursively()
+    }
+
+    @Test
+    fun testThatLogFileDirectoryIsCreatedIfNotExists() {
+        val directory = testProvider.getLogFileDirectory()
+        assert(directory.exists())
+    }
+
+    @Test
+    fun testThatLogFilesListsAllFiles() {
+        val directory = testProvider.getLogFileDirectory()
+        File(directory, UUID.randomUUID().toString()).createNewFile()
+        Assert.assertEquals(testProvider.getLogFiles().count(), 1)
+    }
+
+    @Test
+    fun testThatLogFilesSortsFilesWithMostRecentFirst() {
+        val directory = testProvider.getLogFileDirectory()
+
+        var oldFile = File(directory, UUID.randomUUID().toString())
+        oldFile.createNewFile()
+        oldFile.setLastModified(1_000L)
+        assert(oldFile.lastModified() == 1_000L)
+
+        var newFile = File(directory, UUID.randomUUID().toString())
+        newFile.createNewFile()
+        newFile.setLastModified(1_000_000L)
+        assert(newFile.lastModified() == 1_000_000L)
```

Comment:
AS complains about these being `var`s instead of `val`s. However, there is also a nicer way to write all this:

```suggestion
        listOf(1_000L, 1_000_000L).forEach { modifiedDate ->
            File(directory, UUID.randomUUID().toString()).also { file ->
                file.createNewFile()
                file.setLastModified(modifiedDate)
                assert(file.lastModified() == modifiedDate)
            }
        }
```

Reply:
This works – updated in 1645984007f4980309c8bf45e9b6faba80c54d81.